In [8]:
import mysql.connector

connection = mysql.connector.connect(
    host="mysql-rfam-public.ebi.ac.uk",
    port=4497,
    user="rfamro",
    database="Rfam"
)


a. How many types of tigers can be found in the taxonomy table of the dataset? 


ans: 8 types                                                                                                                                        
Panthera tigris (tiger)                                                                                                                                     
Panthera tigris altaica (Amur tiger)                                                                                                                
Panthera tigris amoyensis (Amoy tiger)                                                                                                              
Panthera tigris corbetti (Indochinese tiger)                                                                                                             
Panthera tigris jacksoni (Malayan tiger)                                                                                                                        
Panthera tigris sondaica (Javan tiger)                                                                                                                      
Panthera tigris sumatrae (Sumatran tiger)                                                                                                           
Panthera tigris tigris (Bengal tiger)

In [2]:
cursor = connection.cursor()
cursor.execute(" SELECT ncbi_id,species FROM taxonomy WHERE species LIKE '%panthera tigris%' ")
rows = cursor.fetchall()
print(len(rows))
for row in rows:
    print(row)


8
(9694, 'Panthera tigris (tiger)')
(74533, 'Panthera tigris altaica (Amur tiger)')
(253258, 'Panthera tigris amoyensis (Amoy tiger)')
(74534, 'Panthera tigris corbetti (Indochinese tiger)')
(419130, 'Panthera tigris jacksoni (Malayan tiger)')
(644771, 'Panthera tigris sondaica (Javan tiger)')
(9695, 'Panthera tigris sumatrae (Sumatran tiger)')
(74535, 'Panthera tigris tigris (Bengal tiger)')


2.What is the "ncbi_id" of the Sumatran Tiger? 


ans: - ncb_id = 9695

b. Find all the columns that can be used to connect the tables in the given database.

Ans:                                                                                                                                                
1.ncbi_id                                                                                                                                           
2.clan_acc                                                                                                                                                  
3.rfam_acc                                                                                                                                              
4.rfamseq_acc                                                                                                                                           
5.auto_wiki

Which type of rice has the longest DNA sequence?

In [7]:
cursor = connection.cursor()
cursor.execute("SELECT t.species, MAX(r.length) FROM taxonomy t INNER JOIN rfamseq r ON t.ncbi_id = r.ncbi_id WHERE t.species LIKE 'Oryza sativa%' GROUP BY t.species; ")
rows = cursor.fetchall()
print(len(rows))
for row in rows:
    print(row)


4
('Oryza sativa (rice)', 339972)
('Oryza sativa endornavirus (isolate Japan)', 13952)
('Oryza sativa Indica Group', 47244934)
('Oryza sativa Japonica Group', 43270923)


ANS : Oryza sativa Indica Group

d.We want to paginate a list of the family names and their longest DNA sequence lengths (in descending order of length) where only families that have DNA sequence lengths greater than 1,000,000 are included. Give a query that will return the 9th page when there are 15 results per page. (hint: we need the family accession ID, family name and the maximum length in the results)

In [9]:
cursor = connection.cursor()
cursor.execute("SELECT t.species,r.accession, MAX(r.length) AS max_length FROM taxonomy t INNER JOIN rfamseq r ON t.ncbi_id = r.ncbi_id WHERE r.length > 1000000 GROUP BY t.species ORDER BY max_length DESC LIMIT 15 OFFSET 120")
rows = cursor.fetchall()
print(len(rows))
for row in rows:
    print(row)


15
('Anopheles gambiae str. PEST', 'AAAB01008807', 23099915)
('Anopheles dirus', 'KB672490', 22947322)
('Hymenolepis microstoma', 'LN906329', 22338478)
('Eutrema salsugineum', 'KI517384', 21755756)
("Propithecus coquereli (Coquerel's sifaka)", 'KQ022624', 21700459)
('Caenorhabditis briggsae', 'FR847112', 21540570)
('Drosophila sechellia', 'CH480815', 21120651)
('Caenorhabditis elegans', 'BX284601', 20924180)
('Sorex araneus (European shrew)', 'JH798183', 20915803)
('Anopheles atroparvus', 'KI421882', 20238125)
('Echinococcus multilocularis', 'LN902841', 20116479)
('Spermophilus parryii (Arctic ground squirrel)', 'QVIC01000005', 20065445)
('Capsella rubella', 'KB870805', 19566579)
('Mustela putorius furo (domestic ferret)', 'GL896918', 17988942)
('Erinaceus europaeus (western European hedgehog)', 'JH835289', 17429815)
